### PR TITLE
Ranking wins (count wins per player)

### PR DESCRIPTION
**Pull request by yzmopolis**
- add a way of calculating player **wins**
- **show wins in ranking page**
- sort the table by wins
- add tests to check the logic

screen shot 2018-06-06 at 17 58 47

### CODE REVIEW

File: app/presenters/web/player_presenter.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ def id
       player.id
     end
 
+    def wins(wins_count = WinsCounter.new)
```
Comment (shadre) [inline app/presenters/web/player_presenter.rb]: It's nitpicky, but I would call the local var `wins_counter` for clarity. It takes a counter (service) and not count (number).

Comment (takiarek) [reply app/presenters/web/player_presenter.rb]: IMO it's important to know what you deal with when you read the name. I would suggest naming variables similarly to the objects they carry.

File: spec/factories/players.rb [hunk 2]
```diff
@@ -20,4 +20,24 @@
...
+  factory :tom, class: "Player" do
+    slack_id "UA2QQD74C"
+    initialize_with { Player.find_or_create_by(slack_id: slack_id) }
+  end
+
+  factory :sherlock, class: "Player" do
+    slack_id "UA2QQD75C"
+    initialize_with { Player.find_or_create_by(slack_id: slack_id) }
+  end
+
```
Comment (takiarek) [inline spec/factories/players.rb]: Why do you need 4 new factories here?

Also, names like this don't tell what kind of object they create. I suggest naming them like `player_one` or `player_zuzia`.

File: spec/services/wins_counter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "factory_bot"
+
+describe WinsCounter do
+  context "correctly counts that" do
```
Comment (takiarek) [inline spec/services/wins_counter_spec.rb]: You don't need to create a context when you only have one :)
Contexts are used to separate tests for code that runs basing on some condition.

File: spec/services/wins_counter_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    let(:peter) { create(:peter) }
+    let(:sherlock) { create(:sherlock) }
+    let(:anna) { create(:anna) }
+    let(:wins_counter) { wins_counter = WinsCounter.new }
+
+    before(:each) do
+      create(:match)
+      create(:match1)
+      create(:match2)
+    end
```
Comment (takiarek) [inline spec/services/wins_counter_spec.rb]: To exhaust all options, I would create one match that ended with a draw, one match that ended with first team win and one where second team won. This way, you will also test that draws are not counted as wins.

**Pull request merged**